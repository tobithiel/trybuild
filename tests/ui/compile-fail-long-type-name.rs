struct Atype<T, K>(T, K);
struct Btype<T, K>(T, K);
struct Ctype<T, K>(T, K);

fn main() {
    let x: Atype<
      Btype<
        Ctype<
          Atype<
            Btype<
              Ctype<
                Atype<
                  Btype<
                    Ctype<i32, i32>,
                    i32
                  >,
                  i32
                >,
                i32
              >,
              i32
            >,
            i32
          >,
          i32
        >,
        i32
      >,
      i32
    > = Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
        Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
            Ok("")
        ))))))))))))))))))))))))))))))
    ))))))))))))))))))))))))))))));
    //~^^^^^ ERROR E0308

    let _ = Some(Ok(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(
        Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(
            Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(
                Some(Some(Some(Some(Some(Some(Some(Some(Some("")))))))))
            )))))))))))))))))
        ))))))))))))))))))
    ))))))))))))))))) == Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
        Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
            Ok(Ok(Ok(Ok(Ok(Ok(Ok("")))))))
        ))))))))))))))))))))))))))))))
    ))))))))))))))))))))))));
    //~^^^^^ ERROR E0308

    let x: Atype<
      Btype<
        Ctype<
          Atype<
            Btype<
              Ctype<
                Atype<
                  Btype<
                    Ctype<i32, i32>,
                    i32
                  >,
                  i32
                >,
                i32
              >,
              i32
            >,
            i32
          >,
          i32
        >,
        i32
      >,
      i32
    > = ();
    //~^ ERROR E0308

    let _: () = Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
        Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
            Ok(Ok(Ok(Ok(Ok(Ok(Ok("")))))))
        ))))))))))))))))))))))))))))))
    ))))))))))))))))))))))));
    //~^^^^^ ERROR E0308
}
