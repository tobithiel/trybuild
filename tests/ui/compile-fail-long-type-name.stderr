error[E0308]: mismatched types
  --> tests/ui/compile-fail-long-type-name.rs:30:9
   |
6  |        let x: Atype<
   |  _____________-
7  | |        Btype<
8  | |          Ctype<
9  | |            Atype<
...  |
29 | |        i32
30 | |      > = Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
   | | _____-___^
   | ||_____|
   |  |     expected due to this
31 |  |         Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
32 |  |             Ok("")
33 |  |         ))))))))))))))))))))))))))))))
34 |  |     ))))))))))))))))))))))))))))));
   |  |__________________________________^ expected struct `Atype`, found enum `Result`
   |
   = note: expected struct `Atype<Btype<Ctype<Atype<Btype<Ctype<Atype<Btype<Ctype<i32, i32>, i32>, i32>, i32>, i32>, i32>, i32>, i32>, i32>`
                found enum `Result<Result<Result<Result<Result<Result<Result<Result<..., ...>, ...>, ...>, ...>, ...>, ...>, ...>, ...>`
           the full type name has been written to '$DIR/target/tests/target/x86_64-unknown-linux-gnu/debug/deps/$CRATE-fc5fb2f065cf8b36.long-type-hash.txt'

error[E0308]: mismatched types
  --> tests/ui/compile-fail-long-type-name.rs:43:26
   |
43 |       ))))))))))))))))) == Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
   |  __________________________^
44 | |         Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
45 | |             Ok(Ok(Ok(Ok(Ok(Ok(Ok("")))))))
46 | |         ))))))))))))))))))))))))))))))
47 | |     ))))))))))))))))))))))));
   | |____________________________^ expected enum `Option`, found enum `Result`
   |
   = note: expected enum `Option<Result<Option<Option<Option<Option<Option<Option<Option<Option<Option<Option<...>>>>>>>>>>, ...>>`
           the full type name has been written to '$DIR/target/tests/target/x86_64-unknown-linux-gnu/debug/deps/$CRATE-fc5fb2f065cf8b36.long-type-hash.txt'
              found enum `Result<Result<Result<Result<Result<Result<Result<Result<..., ...>, ...>, ...>, ...>, ...>, ...>, ...>, ...>`
           the full type name has been written to '$DIR/target/tests/target/x86_64-unknown-linux-gnu/debug/deps/$CRATE-fc5fb2f065cf8b36.long-type-hash.txt'

error[E0308]: mismatched types
  --> tests/ui/compile-fail-long-type-name.rs:74:9
   |
50 |       let x: Atype<
   |  ____________-
51 | |       Btype<
52 | |         Ctype<
53 | |           Atype<
...  |
73 | |       i32
74 | |     > = ();
   | |     -   ^^ expected struct `Atype`, found `()`
   | |_____|
   |       expected due to this
   |
   = note: expected struct `Atype<Btype<Ctype<Atype<Btype<Ctype<Atype<Btype<Ctype<i32, i32>, i32>, i32>, i32>, i32>, i32>, i32>, i32>, i32>`
           found unit type `()`

error[E0308]: mismatched types
  --> tests/ui/compile-fail-long-type-name.rs:77:17
   |
77 |       let _: () = Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
   |  ____________--___^
   | |            |
   | |            expected due to this
78 | |         Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
79 | |             Ok(Ok(Ok(Ok(Ok(Ok(Ok("")))))))
80 | |         ))))))))))))))))))))))))))))))
81 | |     ))))))))))))))))))))))));
   | |____________________________^ expected `()`, found enum `Result`
   |
   = note: expected unit type `()`
                   found enum `Result<Result<Result<Result<Result<Result<Result<Result<..., ...>, ...>, ...>, ...>, ...>, ...>, ...>, ...>`
           the full type name has been written to '$DIR/target/tests/target/x86_64-unknown-linux-gnu/debug/deps/$CRATE-fc5fb2f065cf8b36.long-type-hash.txt'
